Library/Tracking/Koala.swift

```diff
@@ -21,6 +21,24 @@ public final class Koala {
   private var preferredContentSizeCategoryObserver: Any?
   private let screen: UIScreenType
 
+  private enum DataLakeWhiteListedEvents: String, CaseIterable {
```

Comment:
Any reason for the enum vs just an array of event names? I guess an enum could help us in a situation where we want to be sure that same string is used in two places. Should we considering using the `rawValue` of the corresponding case in that event's tracking function too? Something like:

```swift
public func trackDiscovery(params: DiscoveryParams) {
  let props = discoveryProperties(from: params)

  self.track(event: DataLakeWhiteListedEvents.explorePageViewed.rawValue, properties: props)
}
```
🤔 I'm not sure this is what we want, just thinking out loud.

Reply:
Yeah I'd be in favor of doing that. Now's the time to do it too, since we're getting rid of a lot of events. I'm happy to do that in this PR if you're open to it.

Reply:
I think it's nice to have an enum of "clean" events that we can add to, and get auto-complete in the multiple places that we use the event names: in the "track" function, and in the corresponding test. It's less error prone and also easier to re-name, which I've had to do more than once lately 😅

Reply:
Yeah sounds good. Perhaps once we've fully removed the Koala client with plain strings we could change the `track` method signature to receive the `enum` type and only extract the `rawValue` at the point of tracking.

Anyway - one last suggestion before you merge - can you rename it to singular `DataLakeWhiteListedEvent`?